src/dsl/orbit-codegen-dsl/src/main/antlr/cloud/orbit/dsl/Orbit.g4

```diff
@@ -20,11 +20,13 @@ enumDeclaration: ENUM name=ID LC_BRACE members=enumMember* RC_BRACE ;
 enumMember: name=ID EQUAL index=INT SEMI_COLON;
 
 actorDeclaration: ACTOR name=ID LC_BRACE methods=actorMethod* RC_BRACE ;
-actorMethod: returnType=ID name=ID L_PAREN (args=methodParam (COMMA args=methodParam)*)? R_PAREN SEMI_COLON;
-methodParam: type=ID name=ID ;
+actorMethod: returnType=type name=ID L_PAREN (args=methodParam (COMMA args=methodParam)*)? R_PAREN SEMI_COLON;
+methodParam: type name=ID ;
 
 dataDeclaration: DATA name=ID LC_BRACE fields=dataField* RC_BRACE ;
-dataField: type=ID name=ID EQUAL index=INT SEMI_COLON ;
+dataField: type name=ID EQUAL index=INT SEMI_COLON ;
+
+type: name=ID (L_ANGLE of=type (COMMA of=type)* R_ANGLE)? ;
```

Comment:
1. I would add one more level to type: `generic` and `non generic.` It will make extending the grammar easier because there will be cases in which you want just a normal type without supporting generics. 
2. nitpick: consider naming it `typeName` instead of `type`

Reply:
For 2, I'm keeping `type` since in code we already make the distinction between "type" and "type name" (the latter being the Java Poet concept).